### PR TITLE
🐛 issues with docker networking and DNS resolution

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

- Issue1: Using an overlay network for Envoy, will break its internal DNS resolver. Replaced overlay network with normal docker network.
- Issue2: Internal in house DNS domain server is never resolved but Envoy's DNS recover. Removed force search using TCP to solve queries. UDP is the default and what all servers support.
- Other: fixing typos regarding the `envoy` word.

## Related issue/s

- followup to https://github.com/ITISFoundation/osparc-simcore/pull/3606

## Checklist

- [x] Unit tests for the changes exist

### CODE REVIEW

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_egress_config.py [hunk 1]
```diff
@@ -168,27 +170,49 @@ def _add_egress_proxy_network(
...
 ) -> dict[str, Any]:
-
     network_aliases: set[str] = {x[0].hostname for x in egress_proxy_rules}
 
-    envoy_config: dict[str, Any] = _get_envy_config(egress_proxy_rules)
-    yaml_str_envy_config: str = yaml.safe_dump(envoy_config, default_style='"')
-    logger.debug("ENVOY CONFIG\n%s", yaml_str_envy_config)
+    envoy_config: dict[str, Any] = _get_envoy_config(egress_proxy_rules)
+    yaml_str_envoy_config: str = yaml.safe_dump(envoy_config, default_style='"')
+    logger.error("ENVOY CONFIG\n%s", yaml_str_envoy_config)
```
Comment (sanderegg) [inline services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_egress_config.py:206]: error ? that is probably forgotten

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_egress_config.py [hunk 2]
```diff
@@ -197,12 +221,17 @@ def _get_egress_proxy_service_config(
...
         "networks": {
             # allows the proxy to access the internet
-            swarm_network_name: None,
+            network_with_internet: None,
             # allows containers to contact proxy via these aliases
             _get_egress_proxy_network_name(egress_proxy_name): {
                 "aliases": list(network_aliases)
             },
         },
+        # NOTE: there as some issues with ho DNS in house
```
Comment (sanderegg) [inline services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_egress_config.py:230]: you might want to change that comment. in house? like in open source house?

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_egress_config.py [hunk 3]
```diff
@@ -168,27 +170,49 @@ def _add_egress_proxy_network(
...
+    simple_dns_resolver: SimpleDNSResolver,
+) -> list[str]:
+    host_names: deque[str] = deque()
+    queries: deque[Awaitable] = deque()
+
+    for proxy_rule in egress_proxy_rules:
+        data: _HostData = proxy_rule[0]
+        host_names.append(data.hostname)
+        queries.append(
+            simple_dns_resolver.dns_query(
```
Comment (sanderegg) [inline services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_egress_config.py:184]: Q: what happens if this raises?

**Review by sanderegg — CHANGES_REQUESTED** [review]
If I get it right, you are testing these NAT rules before setting them?
Please check my comments.

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_egress_config.py (hunk 2 above)
Comment (GitHK) [reply services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_egress_config.py]: removed, no longer used.

**Comment by GitHK** [comment]
@sanderegg @pcrespov sorry. Just realised this morning what was happening with the DNS resolver. It required UDP and not TCP searching. That's why some servers were not resolving addresses.

File: services/director-v2/requirements/_base.txt [hunk 4]
```diff
@@ -1,41 +1,55 @@
...
 aioredis==2.0.1
     # via aiocache
 aiormq==6.4.2
     # via aio-pika
 aiosignal==1.2.0
     # via aiohttp
 alembic==1.8.1
-    # via -r requirements/../../../packages/postgres-database/requirements/_base.in
+    # via
+    #   -r requirements/../../../packages/postgres-database/requirements/_base.in
```
Comment (pcrespov) [inline services/director-v2/requirements/_base.txt]: Please check that you are using the write version of pip-tools.
To be sure, I recommend you to use the one prepared in the requirements image:
```
cd requirements/tools
make build-nc
make shell
cd ...
cd services/director-v2/requirements
make touch
make reqs target=async-dns
```

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_egress_config.py [hunk 5]
```diff
@@ -168,27 +168,26 @@ def _add_egress_proxy_network(
     service_spec["networks"] = networks
 
 
-def _get_egress_proxy_service_config(
+async def _get_egress_proxy_service_config(
```
Comment (pcrespov) [inline services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_egress_config.py:171]: Why is thisone async ?? pls double check

File: services/director-v2/requirements/_base.in [hunk 6]
```diff
@@ -22,6 +22,7 @@ aio-pika
 aiocache[redis,msgpack]
 aiodocker
 aiopg[sa]
+async-dns
```
Comment (pcrespov) [inline services/director-v2/requirements/_base.in]: Where are you using this library ``async_dns``?

File: services/director-v2/requirements/_base.txt (hunk 4 above)
Comment (GitHK) [reply services/director-v2/requirements/_base.txt]: Those are the exact commands that I am using. I don't know why so much stuff changes.

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_egress_config.py (hunk 5 above)
Comment (GitHK) [reply services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_egress_config.py]: Good catch forgot to revert them

**Review by sanderegg — APPROVED** [review]

**Pull request merged**